tools/cjxl_main.cc

```diff
@@ -683,8 +683,16 @@ void ProcessFlags(const jxl::extras::Codec codec,
                   jxl::extras::JXLCompressParams* params) {
   // Tuning flags.
   ProcessBoolFlag(args->modular, JXL_ENC_FRAME_SETTING_MODULAR, params);
-  ProcessBoolFlag(args->keep_invisible, JXL_ENC_FRAME_SETTING_KEEP_INVISIBLE,
-                  params);
+  if (args->keep_invisible == jxl::Override::kDefault) {
+    int64_t value =
+        (args->distance == 0.0 || args->lossless_jpeg == JXL_TRUE) ? 1 : 0;
+    params->options.emplace_back(JXL_ENC_FRAME_SETTING_KEEP_INVISIBLE, value,
+                                 0);
+  } else {
+    int64_t value = args->keep_invisible == jxl::Override::kOn ? 1 : 0;
+    params->options.emplace_back(JXL_ENC_FRAME_SETTING_KEEP_INVISIBLE, value,
+                                 0);
+  }
```

Comment:
It shouldn't be necessary to duplicate this logic in cjxl — it should already be in libjxl, if JXL_ENC_FRAME_SETTING_KEEP_INVISIBLE gets set to -1 or if it doesn't get set at all.

If libjxl isn't behaving as expected then the bug should get fixed there.

Reply:
Good point.

Changed it around, see what you think.

Reply:
I'm guessing @mo271's approval makes this discussion redundant?